Delegate plot generation and computation to sigrap

qclayssen opened the pull request:
Moves [HRDetect](https://github.com/umccr/sigrap#id_-hrdetect), [MutationalPatterns](https://github.com/umccr/sigrap#id_-mutationalpatterns) figure generation out of **gpgr** and into **sigrap**.  
gpgr now focuses purely on report assembly; all heavy plotting logic lives in [sigrap](https://github.com/umccr/sigrap).
Allows [sash](http://github.com/umccr/sash) to parallel-run MutationalPatterns, HRDetect in separate Nextflow processes.
Add back CHORD plots, input will be read from OA

## Checklist
- [x] End-to-end test on COLO829 sample check.
- [x] Update CHANGELOG.md

[PR#13 sigrap](https://github.com/umccr/sigrap/pull/13)
[PR#16 sigrap](https://github.com/umccr/sash/pull/16)

qclayssen committed:
add params sigrap input files

qclayssen committed:
add rendering reading  sigrap files and associated function

qclayssen committed:
oxygenise and linting

qclayssen committed:
fix mutational_sigs_prep filter and gpgr path

qclayssen committed:
add Dockerfile for sash

qclayssen committed:
Merge branch 'main' into input-sigrap-data

qclayssen committed:
read chord files from OA

qclayssen committed:
change to read sigrap plot and data as inputs

qclayssen committed:
Add changes in News

qclayssen committed:
remove testing docker and path

pdiakumis commented on .dockerignore:
```diff
@@ -1,3 +1,3 @@
-*
```
Any reason for removing the `*`?
https://stackoverflow.com/questions/28097064/dockerignore-ignore-everything-except-a-file-and-the-dockerfile

qclayssen replied:
I made my own docker for testing that needed the directory. I should have revert [the commit]([f9d4](https://github.com/umccr/gpgr/pull/91/commits/f9d4248957d6b89c44171151261a4cd0b7bd7162))

pdiakumis commented on R/rmd.R:
```diff
@@ -31,14 +31,17 @@
                        purple_som_cnv_ann, purple_som_cnv, purple_som_gene_cnv, purple_som_snv_vcf,
                        somatic_snv_vcf, somatic_snv_summary, somatic_sv_tsv, somatic_sv_vcf,
-                       result_outdir, tumor_name, out_file = NULL, quiet = FALSE) {
+                       result_outdir, tumor_name, mutpat_dir=NULL, hrdetect_file=NULL, chord_file=NULL, out_file = NULL, quiet = FALSE) {
```
Consider installing air (https://github.com/posit-dev/air) to handle all the formatting (whitespace, newlines etc.) for you, to make it consistent across the codebase.
You can use `# fmt: skip` if you find that it does too much (https://posit-dev.github.io/air/formatter.html#skip-comments).

qclayssen replied:
I did use air (https://github.com/posit-dev/air) in the [commit](https://github.com/umccr/gpgr/pull/91/commits/0573074f6e57377ad9349e043cd5ade13f017260)

pdiakumis replied:
Unfortunately it sucks with tribbles (https://github.com/posit-dev/air/issues?q=tribble), so you need to put that `# fmt: skip` above them. Can tackle in a future PR, no need to do that now ;-)

qclayssen replied:
I will do it before the release

pdiakumis reviewed (approved):
I haven't looked at the Rmd stuff in detail, but other than the dockerignore file, all LGTM 👍

qclayssen committed:
Uptade using air format .

qclayssen committed:
add back dockerignore *

qclayssen committed:
Merge remote-tracking branch 'origin/main' into input-sigrap-data

qclayssen committed:
remove duplicate argument

qclayssen committed:
devtools check

qclayssen committed:
duplicate line

qclayssen committed:
add back chord in summary

qclayssen committed:
Apply tidyverse/air style via styler (#96)

qclayssen committed:
fix path circos_baf_plot

qclayssen committed:
bump MutationalPatterns and sigrap version

qclayssen committed:
Add config to skippe tibble in toml

qclayssen committed:
linting

qclayssen committed:
Add quentin as as maintainer

qclayssen committed:
linting

qclayssen committed:
linting

qclayssen committed:
update new

qclayssen committed:
use is_hypermutated flag from bolt variant counts JSON

qclayssen committed:
Merge remote-tracking branch 'origin/main' into input-sigrap-data

qclayssen committed:
Hide is_hypermutated from SNVs summary display

The field is used internally to set the Hypermutated flag;
it should not appear in the raw count string shown in the report.
